cpp/src/TrinoExchangeSource.cpp

```diff
@@ -161,61 +252,99 @@ void TrinoExchangeSource::processDataResponse(
     for (auto& buf : iobufs) {
       totalBytes += buf->capacity();
       if (!singleChain) {
-        buf->trimStart(16);
         singleChain = std::move(buf);
       } else {
         singleChain->prev()->appendChain(std::move(buf));
       }
     }
-    TrinoExchangeSource::updateMemoryUsage(totalBytes);
-
-    page = std::make_unique<exec::SerializedPage>(
-        std::move(singleChain), [pool = pool_](folly::IOBuf& iobuf) {
-          int64_t freedBytes{0};
-          // Free the backed memory from MemoryAllocator on page dtor
-          folly::IOBuf* start = &iobuf;
-          auto curr = start;
-          do {
-            freedBytes += curr->capacity();
-            // FIXME: Jiguang
-            // pool->free(curr->writableData(), curr->capacity());
-            curr = curr->next();
-          } while (curr != start);
-          TrinoExchangeSource::updateMemoryUsage(-freedBytes);
-        });
+    // TODO: Update memory
+    // TrinoExchangeSource::updateMemoryUsage(totalBytes);
+
+    VELOX_CHECK_GE(singleChain->length(), TRINO_RESULT_RESPONSE_HEADER_LEN);
+
+    const int32_t magic = *reinterpret_cast<const int32_t*>(singleChain->data());
+    const int64_t checkSum = *reinterpret_cast<const int64_t*>(singleChain->data() + 4);
+    const int32_t pageCount = *reinterpret_cast<const int32_t*>(singleChain->data() + 12);
+
+    VELOX_CHECK_EQ(magic, TRINO_RESULT_RESPONSE_HEADER_MAGIC);
+    VELOX_CHECK_EQ(sequence_ + pageCount, ackSequence);
+    singleChain->trimStart(TRINO_RESULT_RESPONSE_HEADER_LEN);
+
+    int32_t remaingPage = pageCount;
+    int8_t pageHeader[TRINO_SERIALIZED_PAGE_HEADER_SIZE + 4];
+    auto curr = singleChain.get();
+    while (remaingPage) {
+      if (!curr) {
+        VLOG(google::ERROR) << "Received page body in TrinoExchangeSource is corrupted.";
+      }
+
+      readIOBuf(pageHeader, curr, TRINO_SERIALIZED_PAGE_HEADER_SIZE + 4);
+
+      VLOG(1) << fmt::format(
+          "Received Page count: {}, remaining: {}, row_num: {}, uncompressed_size: {}, "
+          "compressed_size: {}, column_num: "
+          "{}, underlay_ptr: {}",
+          pageCount, remaingPage, *reinterpret_cast<int32_t*>(pageHeader),
+          *reinterpret_cast<int32_t*>(pageHeader + 5),
```

Comment:
Let's add some static constants or macro constants to replace this hard core like 5/9/13. such as:

```cpp
static const int32_t kPagePositionCountOffset = 0;
static const int32_t kPageCodecOffset = kPagePositionCountOffset + 4;
static const int32_t kPageUncompressedSizeOffset = kPageCodecOffset + 1;
static const int32_t kPageCompressedSizeOffset = kPageUncompressedSizeOffset + 4;
static const int32_t kPageHeaderLength = kPageCompressedSizeOffset + 4;
```

Reply:
I found there already contains some macro constants above, it's better to add missing constant there.